src/app.ts

```diff
@@ -15,6 +23,60 @@ const findBranchesToProcess = () => {
   return new RegExp(branches)
 }
 
+const processPullRequest = () => async (context: Context<'pull_request'>, app: Probot) => {
+  const payload = context.payload as PullRequestEvent
+  const repository = {
+    owner: payload.repository.owner.login,
+    repo: payload.repository.name,
+  }
+  app.log(`Pull request event happened on '${context.payload.pull_request}'`)
+
+  try {
+    const filesChanged = await context.octokit.pulls.listFiles({
+      ...repository,
+      pull_number: payload.number
+    })
+    const configFile = filesChanged.data.find(x => x.filename === configFileName)
+    if (configFile) {
+
+      app.log.debug(`Found file ${configFile.filename}`)   
+      const checkRun = await createCheckRun(
+        context.octokit,
+        {
+          owner: repository.owner,
+          repo: repository.repo,
+          sha: payload.pull_request.head.sha
+        }
+      )         
+
+      const fileContent = await context.octokit.repos.getContent({
+        ...repository,
+        path: configFile.filename,
+        ref: payload.pull_request.head.ref
+      })
+
+      const { content } = fileContent.data as { content: string }
+      const decodedContent = Buffer.from(content, 'base64').toString()
+      app.log.debug(`decoded content ${decodedContent}`)
+
+
+      const result = validator(schema as JSONSchemaType<TemplateConfig>, decodedContent)
```

Comment:
I'd rip the cast out here to make it explicit that we're casting

```suggestion
      const typedSchema = schema as JSONSchemaType<TemplateConfig>
      const result = validator(typedSchema, decodedContent)
```